From 90a103d6050b266fd8fc8fd0636be32de5885dec Mon Sep 17 00:00:00 2001
From: Jakub Hrozek <jhrozek@redhat.com>
Date: Tue, 25 Oct 2016 16:15:06 +0200
Subject: [PATCH 51/79] CONFDB: Re-enable the files provider
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The files provider was "blacklisted" for a long time, because very old
(pre-1.0) versions of sssd had the capability to create users and groups
by calling into the shadow-utils binaries directly which was later
removed.

Since nobody is (hopefully) running these ancient versions anymore and
we are about to re-enable the files provider, we can remove this check.

Reviewed-by: Pavel Březina <pbrezina@redhat.com>
---
 src/confdb/confdb.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/confdb/confdb.c b/src/confdb/confdb.c
index 69200f5cd4dded1c25dc34e06595809c65670205..b0a75867c61fa3af21990d92419f17baa34864ae 100644
--- a/src/confdb/confdb.c
+++ b/src/confdb/confdb.c
@@ -900,13 +900,6 @@ static int confdb_get_domain_internal(struct confdb_ctx *cdb,
         goto done;
     }
 
-    if (strcasecmp(domain->provider, "files") == 0) {
-        /* The files provider is not valid anymore */
-        DEBUG(SSSDBG_FATAL_FAILURE, "The \"files\" provider is invalid\n");
-        ret = EINVAL;
-        goto done;
-    }
-
     if (strcasecmp(domain->provider, "local") == 0) {
         /* If this is the local provider, we need to ensure that
          * no other provider was specified for other types, since
-- 
2.9.3

